specification.markdown

```diff
@@ -1,66 +1,140 @@
+definition in the rest of the document. To avoid confusion different
+token names has been chosen. The BNFs can be viewed in full in the
+appendices.
+
+## EBML principle
+
+EBML specifies a binary and octet (byte) aligned format inspired by the
+principle of XML. EBML itself is a generalized description of the
+technique of binary markup. Like XML, it is completely agnostic to any
+data that it might contain. The format is made of 2 parts: the semantic
+and the syntax. The semantic specifies a number of Element IDs and their basic type and is not included in the data file/stream.
+
+Just like XML, the specific "tags" (Element IDs in EBML parlance) used in an EBML implementation are arbitrary. However, the semantic of EBML
+outlines general data types and Element IDs.
 
 The known basic types are:
 
--   Signed Integer - Big-endian, any size from 1 to 8 octets
--   Unsigned Integer - Big-endian, any size from 1 to 8 octets
--   Float - Big-endian, defined for 4 and 8 octets (32, 64 bits)
--   String - Printable ASCII (0x20 to 0x7E), zero-padded when needed
--   UTF-8 - [Unicode string](http://www.unicode.org/), zero padded when
-    needed ([RFC 2279](http://www.faqs.org/rfcs/rfc2279.html))
--   Date - signed 8 octets integer in nanoseconds with 0 indicating the
-    precise beginning of the millennium (at
-    2001-01-01T00:00:00,000000000 UTC)
--   master-element - contains other EBML sub-elements of the next lower
-    level
--   Binary - not interpreted by the parser
+Data Type        | Definition
+-----------------|----------------------------------------------------
+Signed Integer   | Big-endian, any size from 1 to 8 octets
+Unsigned Integer | Big-endian, any size from 1 to 8 octets
+Float            | Big-endian, defined for 4 and 8 octets (32, 64
+                 | bits)
+String           | Printable ASCII (0x20 to 0x7E), zero-padded when
+                 | needed
+UTF-8            | [Unicode string](http://www.unicode.org/), zero
+                 | padded when needed ([RFC 2279](http://www.faqs.org/rfcs/rfc2279.html))
+Date             | signed 8 octets integer in nanoseconds with 0
+                 | indicating the precise beginning of the
+                 | millennium (at 2001-01-01T00:00:00.000000000 UTC)
+Master-element   | contains other EBML sub-elements of the next lower
+                 | level
+Binary           | not interpreted by the parser
 
 As well as defining standard data types, EBML uses a system of Elements
-to make up an EBML "document." Elements incorporate an Element ID, a
-descriptor for the size of the element, and the binary data itself.
-Futher, Elements can be nested, or contain, Elements of a lower "level."
+to make up an EBML "document". Elements incorporate an Element ID, a
+descriptor for the size of the element (known as the Element Data
+Size), and the binary data itself. Futher, Elements can be nested or
+contain Elements of a lower "level".
```

Comment:
Add an issue for now, back to this later: https://github.com/Matroska-Org/ebml-specification/issues/7